CRM/Contact/Form/Task/EmailCommon.php

```diff
@@ -88,43 +104,48 @@ public static function preProcessFromAddress(&$form) {
           $form->_noEmails = FALSE;
         }
       }
+      if (!empty($email)) {
+        $form->_emails[$emailId] = $emails[$emailId];
+        $emails[$emailId] .= $item['locationType'];
 
-      $form->_emails[$emailId] = $emails[$emailId];
-      $emails[$emailId] .= $item['locationType'];
-
-      if ($item['is_primary']) {
-        $emails[$emailId] .= ' ' . ts('(preferred)');
+        if ($item['is_primary']) {
+          $emails[$emailId] .= ' ' . ts('(preferred)');
+        }
+        $emails[$emailId] = htmlspecialchars($emails[$emailId]);
       }
-      $emails[$emailId] = htmlspecialchars($emails[$emailId]);
     }
 
     $form->assign('noEmails', $form->_noEmails);
 
-    if ($form->_noEmails) {
-      CRM_Core_Error::statusBounce(ts('Your user record does not have a valid email address'));
+    if ($bounce) {
+      if ($form->_noEmails) {
+        CRM_Core_Error::statusBounce(ts('Your user record does not have a valid email address'));
+      }
     }
 
     // now add domain from addresses
-    $domainEmails = array();
-    $domainFrom = CRM_Core_OptionGroup::values('from_email_address');
-    foreach (array_keys($domainFrom) as $k) {
-      $domainEmail = $domainFrom[$k];
-      $domainEmails[$domainEmail] = htmlspecialchars($domainEmail);
+    $domainEmails = self::domainEmails();
+    foreach ($domainEmails as $domainEmail => $email) {
       $form->_emails[$domainEmail] = $domainEmail;
     }
-
     $form->_fromEmails = CRM_Utils_Array::crmArrayMerge($emails, $domainEmails);
-
-    // Add signature
-    $defaultEmail = civicrm_api3('email', 'getsingle', array('id' => key($form->_fromEmails)));
-    $defaults = array();
-    if (!empty($defaultEmail['signature_html'])) {
-      $defaults['html_message'] = '<br/><br/>--' . $defaultEmail['signature_html'];
+    foreach ($form->_fromEmails as $key => $fromEmail) {
+      if (empty($fromEmail)) {
+        unset($form->_fromEmails[$key]);
+      }
     }
-    if (!empty($defaultEmail['signature_text'])) {
-      $defaults['text_message'] = "\n\n--\n" . $defaultEmail['signature_text'];
+    if (!empty($emails)) {
+      // Add signature
+      $defaultEmail = civicrm_api3('email', 'getsingle', array('id' => key($form->_fromEmails)));
```

Comment:
This throws an exception(check JIRA for reproducing) when `key($form->_fromEmails)` evaluates to `FALSE`. As key() returns the index element of the **current** array position, this current pointer [return a boolean FALSE](http://php.net/manual/en/function.current.php) after the array is being looped.

Hence, as the foreach() just targets to remove the empty values from an array, replacing it with an array functions should keep the pointer unaffected. Something like -

    $form->_fromEmails = array_filter($form->_fromEmails);

OR `reset($form->_fromEmails)` after the loop. Checking for integer value before calling the api would avoid additional exceptions to be thrown as domainEmails() function return string as a key.